### PR TITLE
Marketplace fixes for api gateway integration

### PR DESCRIPTION
**Pull request by mservidio**
- Addresses open api spec issues for api gateway.
- Adds support to call docs without a token.
- Listener service now installs to managed cloud run.

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [X] Does your submission pass tests?
2. [X] Have you lint your code locally prior to submission?

### CODE REVIEW

File: api/v1/src/accounts/index.js [hunk 1]
```diff
@@ -682,11 +742,260 @@ accounts.get(['/projects/:projectId/accounts:register', '/accounts:register'], a
...
+ *             type: "string"
+ *           Access-Control-Allow-Methods:
+ *             type: "string"
+ *           Access-Control-Allow-Origin:
+ *             type: "string"
+ *   post:
+ *     summary: Register a marketplace account based off request body
+ *     description: Returns a redirect response
+ *     operationId: registerAccount
+ *     security: [] # no security for preflight requests
```
Comment (mservidio) [inline api/v1/src/accounts/index.js:770]: This POST is invoked by a non-logged in user therefore no security should be enabled here.

File: api/v1/src/accounts/index.js [hunk 2]
```diff
@@ -653,8 +653,69 @@ accounts.get('/datasets/:datasetId/accounts', async (req, res) => {
...
+ *             type: "string"
+ *           Access-Control-Allow-Methods:
+ *             type: "string"
+ *           Access-Control-Allow-Origin:
+ *             type: "string"
+ *   get:
+ *     summary: Register a marketplace account based off request body
+ *     description: Returns a redirect response
+ *     operationId: registerAccountGet
+ *     security: [] # no security for marketplace register get request
```
Comment (mservidio) [inline api/v1/src/accounts/index.js:681]: GET should have no security as it's invoked with no token from marketplace.

File: api/v1/src/accounts/index.js [hunk 3]
```diff
@@ -682,11 +743,192 @@ accounts.get(['/projects/:projectId/accounts:register', '/accounts:register'], a
...
-// Backwards compatibility for marketplace
+/**
+ * @swagger
+ *
+ * /accounts:register:
+ *   post:
+ *     summary: Register a marketplace account based off request body
+ *     description: Returns a redirect response
+ *     operationId: registerAccount
+ *     security: [] # no security for marketplace register post request
```
Comment (mservidio) [inline api/v1/src/accounts/index.js:754]: POST should have no security as it's invoked with no token from marketplace.

File: api/v1/src/accounts/index.js [hunk 4]
```diff
@@ -682,11 +743,207 @@ accounts.get(['/projects/:projectId/accounts:register', '/accounts:register'], a
...
+ *       500:
+ *         description: Error
+ *         schema:
+ *           $ref: '#/definitions/Error'
+ */
+/**
+ * @swagger
+ *
+ * /accounts:reset:
+ *   options:
```
Comment (mservidio) [inline api/v1/src/accounts/index.js:841]: I don't understand yet why this options is needed, though without it API gateway denies the post.

File: api/v1/src/accounts/index.js [hunk 5]
```diff
@@ -682,11 +743,241 @@ accounts.get(['/projects/:projectId/accounts:register', '/accounts:register'], a
...
+ *       500:
+ *         description: Error
+ *         schema:
+ *           $ref: '#/definitions/Error'
+ */
+/**
+ * @swagger
+ *
+ * /accounts:syncMarketplace:
+ *   options:
```
Comment (mservidio) [inline api/v1/src/accounts/index.js:920]: I don't understand yet why this options is needed, though without it API gateway denies the post.

File: api/v1/src/accounts/index.js [hunk 6]
```diff
@@ -682,11 +743,241 @@ accounts.get(['/projects/:projectId/accounts:register', '/accounts:register'], a
...
+ *       500:
+ *         description: Error
+ *         schema:
+ *           $ref: '#/definitions/Error'
+ */
+/**
+ * @swagger
+ *
+ * /accounts:activate:
+ *   options:
```
Comment (mservidio) [inline api/v1/src/accounts/index.js:796]: I don't understand yet why this options is needed, though without it API gateway denies the post.

**Review by swilliams11 — APPROVED** [review]
LGTM

**Pull request merged**